src/controlplane/scraper.go

```diff
@@ -29,13 +29,15 @@ type Providers struct {
 // Scraper takes care of getting metrics all control plane instances based on the configuration.
 type Scraper struct {
 	Providers
-	logger               log.Logger
-	config               *config.Config
-	k8sVersion           *version.Info
-	components           []component
-	informerClosers      []chan<- struct{}
-	podListerByNamespace map[string]v1.PodNamespaceLister
-	inClusterConfig      *rest.Config
+	logger                  log.Logger
+	config                  *config.Config
+	k8sVersion              *version.Info
+	components              []component
+	informerClosers         []chan<- struct{}
+	podListerByNamespace    map[string]v1.PodNamespaceLister
+	secretListerByNamespace map[string]v1.SecretNamespaceLister
```

Comment:
I would not save this into the scraper if possible

Reply:
resolve this comment, is part of the discussion above